test/lib/mayaUsd/fileio/testAddMayaReference.py

```diff
@@ -0,0 +1,121 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import fixturesUtils
+
+import mayaUsd
+import mayaUtils
+
+from maya import cmds
+from maya import standalone
+from mayaUsdLibRegisterStrings import getMayaUsdLibString
+from mayaUSDRegisterStrings import getMayaUsdString
+
+import unittest
+
+class AddMayaReferenceTestCase(unittest.TestCase):
+    '''Test Add Maya Reference.
+    '''
+    pluginsLoaded = False
+
+    @classmethod
+    def setUpClass(cls):
+        fixturesUtils.readOnlySetUpClass(__file__, loadPlugin=False)
+        if not cls.pluginsLoaded:
+            cls.pluginsLoaded = mayaUtils.isMayaUsdPluginLoaded()
+
+    @classmethod
+    def tearDownClass(cls):
+        standalone.uninitialize()
+
+    def setUp(self):
+        cmds.file(new=True, force=True)
+
+    def createSimpleMayaScene(self):
+        import os
+        import maya.cmds as cmds
+        import tempfile
+
+        cmds.CreatePolygonSphere()
+        tempMayaFile = os.path.join(tempfile.gettempdir(), 'simpleSphere.ma')
+        cmds.file(rename=tempMayaFile)
+        cmds.file(save=True, force=True, type='mayaAscii')
+        return tempMayaFile
+
+    def testDefineInVariant(self):
+        '''Test the "Define in Variant" options for Add Maya Reference.'''
+
+        # First we need a pure Maya scene to reference in.
+        mayaSceneStr = self.createSimpleMayaScene()
+
+        # Then create an Xform and add the maya reference.
+        cmds.file(new=True, force=True)
+        import mayaUsd_createStageWithNewLayer
+        proxyShapePathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+        stage = mayaUsd.lib.GetPrim(proxyShapePathStr).GetStage()
+        primA = stage.DefinePrim('/A', 'Xform')
+        primPathStr = proxyShapePathStr + ',/A'
+
+        # Since this is a brand new prim, it should not have variant sets.
+        primA = mayaUsd.ufe.ufePathToPrim(primPathStr)
+        self.assertFalse(primA.HasVariantSets())
+
+        kDefaultVariantSetName = getMayaUsdLibString('kMayaRefDefaultVariantSetName')
+        kDefaultVariantName = getMayaUsdLibString('kMayaRefDefaultVariantName')
+
+        import mayaUsdAddMayaReference
+        mayaUsdAddMayaReference.createMayaReferencePrim(
+            primPathStr,
+            mayaSceneStr,
+            'simpleSphere', kDefaultVariantSetName, kDefaultVariantName)
+
+        # Make sure the prim has the variant set and variant.
+        self.assertTrue(primA.HasVariantSets())
+        vset = primA.GetVariantSet(kDefaultVariantSetName)
+        self.assertTrue(vset.IsValid())
+        self.assertEqual(vset.GetName(), kDefaultVariantSetName)
+        self.assertTrue(vset.GetVariantNames())
+        self.assertTrue(vset.HasAuthoredVariant(kDefaultVariantName))
+        self.assertEqual(vset.GetVariantSelection(), kDefaultVariantName)
+
```

Comment:
Would be good to actually check that the Maya reference prim is inside the variant, and that it has the expected metadata.